src/account/operations/helpers/time.rs

```diff
@@ -13,42 +13,17 @@ pub(crate) fn can_output_be_unlocked_now(
     alias_and_nft_addresses: &[Address],
     output_data: &OutputData,
     current_time: u32,
+    alias_state_transition: bool,
 ) -> bool {
-    if let Some(unlock_conditions) = output_data.output.unlock_conditions() {
-        if unlock_conditions.is_time_locked(current_time) {
-            return false;
-        }
-
-        if let Some(address_unlock_condition) = unlock_conditions.address() {
-            let address = address_unlock_condition.address();
-
-            let unlock_address = unlock_conditions.locked_address(address, current_time);
-            // The address that can unlock the output needs to belong to the account
-            if !account_addresses.iter().any(|a| a.address.inner == *unlock_address)
-                && !alias_and_nft_addresses.contains(unlock_address)
-            {
-                return false;
-            };
-        }
-
-        match &output_data.output {
-            Output::Alias(alias_output) => {
-                alias_and_nft_addresses.contains(alias_output.governor_address())
-                    || alias_and_nft_addresses.contains(alias_output.state_controller_address())
-                    || account_addresses.iter().any(|a| {
-                        a.address.inner == *alias_output.governor_address()
-                            || a.address.inner == *alias_output.state_controller_address()
-                    })
-            }
-            Output::Foundry(foundry_output) => {
-                alias_and_nft_addresses.contains(&Address::Alias(*foundry_output.alias_address()))
-            }
-            // Checked these cases before
-            _ => true,
-        }
-    } else {
-        false
-    }
+    let (required_unlock_address, _unlocked_alias_or_nft_address) = output_data
+        .output
+        .required_and_unlocked_address(current_time, output_data.output_id, alias_state_transition)
+        .expect("treasury output is not supported");
```

Comment:
Is that always going to be the only possible error 🤔 ?

Reply:
At least now, I could do unwrap_or(false) or return the error what do you prefer?

Reply:
Bit annoying but I would return the error